tests/EntityFrameworkCore.Projectables.FunctionalTests/MemberBodyParameterValueTests.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.ComponentModel.DataAnnotations.Schema;
+using System.Linq;
+using System.Linq.Expressions;
+using EntityFrameworkCore.Projectables.FunctionalTests.Helpers;
+using Microsoft.EntityFrameworkCore;
+using Xunit;
+
+namespace EntityFrameworkCore.Projectables.FunctionalTests
+{
+    public class MemberBodyParameterValueTests
+    {
+        public class Order
+        {
+            [Key, DatabaseGenerated(DatabaseGeneratedOption.None)]
+            public int Id { get; set; }
+            public List<OrderItem> OrderItem { get; set; } = new List<OrderItem>();
+
+            [Projectable(nameof(GetOrderItemNameExpression), 1)]
```

Comment:
This example confuses me, could you not just have done:

```csharp
[Projectable(UseMemberBody = nameof(GetOrderItemNameInnerExpressionWithArgument))]
public string FirstOrderPropName => GetOrderItemName(1);

private static Expression<Func<Order, string>> GetOrderItemNameInnerExpressionWithArgument()
   => new MyParameterExpressionVisitor(parameter: "id", value: 1).Visit(GetOrderItemNameInnerExpression());

private static Expression<Func<Order, int, string>> GetOrderItemNameInnerExpression()
    => (@this, id) => @this.OrderItem
        .Where(av => av.Id == id)
        .Select(av => av.Name)
        .FirstOrDefault() ?? string.Empty;
```

Where `MyParameterExpressionVisitor` swaps out the parameter named id for a constant value.

Reply:
I'm also confused, but we generate this call in runtime and therefore it is convenient for us to pass the value of the parameters through the attribute.